### ISSUE TITLE
Check for SQL injection

### ISSUE THREAD

**Issue by jhakala** [HCALRunControl/RunInfoDiffer#1]
It seems the code allows SQL injection. For example, one can insert additional SQL into the run numbers:
e.g. 
http://hcalmon.cms/cgi-bin/RunInfoDiffer/viewDiffer.py?runnumber1=300363&runnumber2=300364
versus
http://hcalmon.cms/cgi-bin/RunInfoDiffer/viewDiffer.py?runnumber1=300363%20AND%20name=%27CMS.HCAL_HBHEc:CFGDOC_TXT%27&runnumber2=300364%20AND%20name=%27CMS.HCAL_HBHEc:CFGDOC_TXT%27

This example is relatively harmless, but in principle SQL injection could cause problems in the web server of the app as well as in the database backend.

**Comment by CiaranGodfrey** [comment]
Would this still be possible if I changed from get to post for HTTP method. SQL injection doesn't work through the actual interface, if I removed the info from the URL would there still be a way to do it?

**Comment by jhakala** [comment]
I'm not sure if it would make a difference if you changed from get to post -- I'd bet it would still be possible. (And FWIW, the interface only puts a browser-level protection against the SQL injection -- the html5 input field wants an integer, assuming you have a cooperative browser, but it's not hard to avoid... see the below screenshot).

 Probably the better way to do this is to validate the input from the form. E.g., you could check whether the run number supplied is valid here: https://github.com/HCALRunControl/RunInfoDiffer/blob/master/viewDiffer.py#L35

![image](https://user-images.githubusercontent.com/8178838/29042959-cbe1081c-7b7e-11e7-83e0-e4c1493b7cde.png)

**Comment by kakwok** [comment]
I agree we should validate the input, which should be easy and effective.

**Comment by CiaranGodfrey** [comment]
Okay I'll do that

**Comment by CiaranGodfrey** [comment]
I believe this is solved can I close this issue

**Comment by jhakala** [comment]
Could a check that run number > 0 be added? The database has some issues with run number 0; there is a lot of junk in the database with the [fake] run number 0 and it takes a very long time to query it.
